internal/backends/collection.go

```diff
@@ -115,7 +115,7 @@ type UpdateResult struct {
 // Update updates documents in collection.
 func (cc *collectionContract) Update(ctx context.Context, params *UpdateParams) (res *UpdateResult, err error) {
 	defer observability.FuncCall(ctx)()
-	defer checkError(err)
+	defer checkError(err, ErrorCodeCollectionDoesNotExist)
```

Comment:
Why? I don't see where we handle that error, and updating non-existing collections is possible in MongoDB

Reply:
From my understanding `sqlite` msg_update handles non-existent collections. So when this function is called we expect that collection already exists.

About the error code: we create contracts for backend codes only if they are handled somewhere?

Reply:
I mean, we should not create error codes if they are ignored. The same thing with `Query` for non-existing databases or collections – primary users (`find`, `aggregate`, etc) just return empty results, so why would we make it return an error just to be ignored?

Reply:
For now we are able to easily see where and how functions like `Query` are used, although when we'll start to add more and more functionality the `Query` function might be used in many places, but someone might make a mistake and forget to ensure that collection exists beforewards. In results one would need to spend some time on debugging empty `Query` results. If we return the `lazyerror` that we expect to not be reached we can easily see what went wrong and how we can fix it.

If our current design assumes that we don't return such errors then let's just return empty results, although I'd suggest to consider this later